apsuite/commisslib/measure_bbb_data.py

```
@@ -523,3 +523,250 @@ def waterfall_plot(axis, xs, zs, data):
+                return True
+        return False
+
+    def run_meas(self, save=True):
+        """."""
+        data = dict()
+        datah = list()
+        datav = list()
+        currs = list()
+        for curr in self.params.currents:
+            t0 = _time.time()
+            self.inject_in_storage_ring(curr)
+
+            trial = 0
+            while self._check_pingers_problem():
+                if trial > 2:
+                    print('3 unsucessful reset trials. Exiting...')
+                    break
+                print('Problem with pingers voltage. Resetting...')
+                trial += 1
+
+            _time.sleep(self.params.wait_bbb)
+            print('Acquiring data...')
+            currs.append(self.devices['bbbh'].dcct.current)
+            data['stored_current'] = currs
+            if 'H' in self.params.plane:
+                datah.append(self.get_data(plane='H'))
+                data['horizontal'] = datah
+            if 'V' in self.params.plane:
+                datav.append(self.get_data(plane='V'))
+                data['vertical'] = datav
+            self.data = data
+            if save:
+                self.save_data(fname=self.params.filename, overwrite=True)
+                print('Data saved!')
+            tf = _time.time()
+            print(f'Elapsed time: {tf-t0:.2f}s \n')
+
+    def plot_spectrum(
+            self, plane, freq_min=None, freq_max=None,
+            title=None, fname=None):
+        """plane: must be 'H' or 'V'."""
+        fig = _mplt.figure(figsize=(8, 6))
+        gs = _mgs.GridSpec(1, 1)
+        ax = fig.add_subplot(gs[0, 0])
+
+        curr = _np.array(self.data['stored_current'])
+        if plane.upper() == 'H':
+            data = self.data['horizontal']
+            freq_min = freq_min or 38
+            freq_max = freq_max or 52
+        if plane.upper() == 'V':
+            data = self.data['vertical']
+            freq_min = freq_min or 72
+            freq_max = freq_max or 84
+
```

I think we should return an error here in case plane is not what we expect it to be.

---

Yes, it is better. Done!